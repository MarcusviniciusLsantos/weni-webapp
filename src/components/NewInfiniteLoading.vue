<template>
  <div
    v-show="!complete"
    ref="infinite-loading-element"
  >
    <slot />
  </div>
</template>

<script>
export default {
  props: {
    value: {
      type: Boolean,
    },

    complete: {
      type: Boolean,
    },
  },

  data() {
    return {
      intersectionObserver: null,
    };
  },

  mounted() {
    this.intersectionObserver = new IntersectionObserver((entries) => {
      entries.forEach((entry) => {
        this.$emit('update:model-value', entry.isIntersecting);
      });
    });

    this.intersectionObserver.observe(this.$refs['infinite-loading-element']);
  },

  beforeUnmount() {
    this.intersectionObserver.unobserve(this.$refs['infinite-loading-element']);
  },
};
</script>
