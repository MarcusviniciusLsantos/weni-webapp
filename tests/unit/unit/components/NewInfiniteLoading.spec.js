import { vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import NewInfiniteLoading from '@/components/NewInfiniteLoading.vue';

describe('NewInfiniteLoading.vue', () => {
  let wrapper;

  beforeEach(() => {
    const observe = vi.fn();
    const unobserve = vi.fn();

    // you can also pass the mock implementation
    // to vi.fn as an argument
    window.IntersectionObserver = vi.fn(() => ({
      observe,
      unobserve,
    }));

    wrapper = shallowMount(NewInfiniteLoading, {});
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
