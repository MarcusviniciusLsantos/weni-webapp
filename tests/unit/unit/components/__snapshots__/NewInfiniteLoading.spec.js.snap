// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewInfiniteLoading.vue > renders correctly 1`] = `<div />`;
